### PR TITLE
Confirm dialog

### PR DESCRIPTION
**Pull request by colinyoung87**

### CODE REVIEW

File: app/assets/javascripts/features/confirm_dialog.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+$(function(){
+  $(document).on('click', '.confirm-dialog-wrapper .close-dialog', function(ev){
+    $(this).closest('.confirm-dialog-wrapper').fadeOut(100);
```
Comment (phawk) [inline app/assets/javascripts/features/confirm_dialog.js]: Do we need to remove it from the DOM after it’s faded out?

Comment (phawk) [inline app/assets/javascripts/features/confirm_dialog.js]: ah nevermind, seen the rest of the code doh.

File: app/assets/javascripts/features/confirm_dialog.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  if (link.data("confirm") == undefined){
+    return true;
+  }
+  $.rails.showConfirmationDialog(link);
+  return false;
+}
+
+// Display the confirmation dialog
+$.rails.showConfirmationDialog = function(link){
+  var message = link.data("confirm");
```
Comment (phawk) [inline app/assets/javascripts/features/confirm_dialog.js]: Wonder can we add an optional data-confirm-title to links, if it’s present use it as the title and then data-confirm as the message. If it’s not present fallback to 'Are you sure?'

**Comment by phawk** [comment]
:+1:

File: app/assets/javascripts/features/confirm_dialog.js (hunk 1 above)
Comment (colinyoung87) [inline app/assets/javascripts/features/confirm_dialog.js]: Could use a JS templates to build it each time, and then remove it after it's finished if you think that'd be a nicer way of handling it?

Comment (phawk) [inline app/assets/javascripts/features/confirm_dialog.js]: Ack no this is fine, just hadn’t read the rest of the code before I just read this bit

File: app/assets/javascripts/features/confirm_dialog.js (hunk 2 above)
Comment (colinyoung87) [inline app/assets/javascripts/features/confirm_dialog.js]: How do you mean? Is it not always going to need the data-confirm as a message anyway? We could maybe have an optional field for updating the text on the confirm button itself?

**Pull request merged**